crates/cdk/src/publish.rs

```diff
@@ -106,6 +102,22 @@ impl PublishCmd {
             }
         }
 
+        if !self.pack {
+            self.cleanup(&package_info)?;
+        }
```

Comment:
Instead of calling separately, we skip running the `cleanup` if the `--pack` argument is not present.
This way we always cleanup unless specified.